src/main/java/com/example/bookstoreapp/security/JwtAuthenticationFilter.java

```diff
@@ -0,0 +1,57 @@
+package com.example.bookstoreapp.security;
+
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+@RequiredArgsConstructor
+@Component
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+
+    public static final int BEGIN_INDEX = 7;
+    public static final String START_WITH_BEARER_ = "Bearer ";
```

Comment:
```suggestion
    public static final String START_WITH_BEARER = "Bearer ";
```
you can use START_WITH_BEARER.length instead of BEGIN_INDEX